elrond-wasm-debug/src/testing_framework/mandos_generator.rs

```diff
@@ -0,0 +1,122 @@
+use std::{collections::BTreeMap, fs::File, io::Write};
+
+use elrond_wasm::types::Address;
+use mandos::serde_raw::{ScenarioRaw, StepRaw};
+use serde::Serialize;
+
+use super::{raw_converter::*, ScCallMandos, ScQueryMandos, TxExpectMandos};
+use crate::world_mock::{AccountData, BlockInfo};
+
+pub(crate) struct MandosGenerator {
+    scenario: ScenarioRaw,
+    current_tx_id: u64,
+}
+
+impl MandosGenerator {
+    pub fn new() -> Self {
+        Self {
+            scenario: ScenarioRaw {
+                check_gas: None,
+                comment: None,
+                gas_schedule: None,
+                name: None,
+                steps: Vec::new(),
+            },
+            current_tx_id: 0,
+        }
+    }
+
+    pub fn write_mandos_output(self, file_path: &str) {
+        let buf = Vec::new();
+        let formatter = serde_json::ser::PrettyFormatter::with_indent(b"    ");
```

Comment:
We could extract this logic somewhere, I think it is also duplicated in some other tests.

Reply:
I copied this from the serde test, so yeah, it's duplicated from there.